### PR TITLE
[rlgl] Check for extensions before enabling them

### PR DESCRIPTION
**Pull request by Not-Nik**
Updates `external/glad.h` with the new required extension macros.
Draft because I couldn't get it to build on my machine.
Addresses #2695

### CODE REVIEW

**Comment by Not-Nik** [comment]
Oh yea, I'll need to do extra work on macOS as well...

**Comment by Not-Nik** [comment]
Unsure whats with the Windows build, I'll look into that later

**Comment by Not-Nik** [comment]
Note to self: ASTC compression might not be present in all OGL 3.3 implementations

File: src/external/glad.h [hunk 1]
```diff
@@ -8304,7 +8336,7 @@ static int glad_gl_get_extensions( int version, const char **out_exts, unsigned
             const char *gl_str_tmp = (const char*) glad_glGetStringi(GL_EXTENSIONS, index);
             size_t len = strlen(gl_str_tmp) + 1;
 
-            char *local_str = (char*) GLAD_MALLOC(len * sizeof(char));
```
Comment (raysan5) [inline src/external/glad.h:8307]: It should use `GLAD_MALLOC` and `GLAD_FREE`

File: src/external/glad.h [hunk 2]
```diff
@@ -12,16 +12,16 @@
  *  - ALIAS = False
  *  - DEBUG = False
  *  - HEADER_ONLY = True
- *  - LOADER = True
+ *  - LOADER = False
```
Comment (raysan5) [inline src/external/glad.h:15]: I think that statement is not correct (?)

File: src/rlgl.h [hunk 3]
```diff
@@ -2018,28 +2011,32 @@ void rlLoadExtensions(void *loader)
     for (int i = 0; i < numExt; i++) TRACELOG(RL_LOG_INFO, "    %s", glGetStringi(GL_EXTENSIONS, i));
 #endif
 
+#if defined(GRAPHICS_API_OPENGL_21)
```
Comment (raysan5) [inline src/rlgl.h:2014]: This code structure looks a bit complex. I prefer to just use a `#if defined(GRAPHICS_API_OPENGL_21)` check for the required extensions and duplicate code, it's more visual.

File: src/external/glad.h (hunk 2 above)
Comment (Not-Nik) [reply src/external/glad.h:15]: Yes it is. raylib doesn't use glad's loader function, but instead `glfwGetProcAddress`. Using glad's loader requires inclusion of `windows.h` which introduces name conflicts for some functions.

File: src/rlgl.h [hunk 4]
```diff
@@ -2028,18 +2033,18 @@ void rlLoadExtensions(void *loader)
...
-    #if defined(__APPLE__)
-    // Apple provides its own extension macros
-    RLGL.ExtSupported.texCompDXT = GL_EXT_texture_compression_s3tc;
-    #else
-    // NOTE: With GLAD, we can check if an extension is supported using the GLAD_GL_xxx booleans
-    RLGL.ExtSupported.texCompDXT = GLAD_GL_EXT_texture_compression_s3tc;  // Texture compression: DXT
-    RLGL.ExtSupported.texCompETC2 = GLAD_GL_ARB_ES3_compatibility;        // Texture compression: ETC2/EAC
-    #endif
+
+#undef CORE_OPENGL_33
```
Comment (raysan5) [inline src/rlgl.h:2047]: No need for this undef

**Comment by raysan5** [comment]
@Not-Nik Looks great! Thanks for the improvement and all the reviews!

**Pull request merged**